packages/collaboration/src/sharedlink.ts

```diff
@@ -45,22 +45,458 @@ export async function showSharedLinkDialog({
+            ) ||
+            userData.scopes.includes('admin:servers')
+          ) {
+            canControlServer = true;
+          }
+        }
+      }
+    }
+  }
+
+  // If we can create a share, open the proper UI
+  if (canCreateShare) {
+    let readableServerName = serverName;
+    if (readableServerName === '') {
+      readableServerName = 'default';
+    }
+    return showDialog({
+      title: trans.__('Share Jupyter Server ') + readableServerName,
+      body: new ManageSharesBody(
+        url.toString(),
+        serverName,
+        canListUsers,
+        canListGroups,
+        canControlServer,
+        serverOwner,
+        hubApiUrl,
+        token,
+        trans
+      ),
+      buttons: [
+        Dialog.cancelButton({
+          label: trans.__('Close'),
+          caption: trans.__('Close the dialog')
+        }),
+        Dialog.okButton({
+          label: trans.__('Copy Link'),
+          caption: trans.__('Copy the link to the Jupyter Server')
+        })
+      ]
+    });
+    // If we can't create a real share, we show the legacy dialog that just copies the URL with the server owner's token
+  } else {
+    return showDialog({
+      title: trans.__('Share Jupyter Server Link'),
+      body: new SharedLinkBody(
+        url.toString(),
+        token,
+        PageConfig.getOption('hubUser') !== '',
+        trans
+      ),
+      buttons: [
+        Dialog.cancelButton(),
+        Dialog.okButton({
+          label: trans.__('Copy Link'),
+          caption: trans.__('Copy the link to the Jupyter Server')
+        })
+      ]
+    });
+  }
+}
+
+class ManageSharesBody extends Widget implements Dialog.IBodyWidget {
+  private _users: any[] = [];
+  private _shares: any[] = [];
+  private _searchInput: HTMLInputElement | null = null;
+  private _searchResults: HTMLDivElement | null = null;
+  private _sharesContainer: HTMLDivElement | null = null;
+
+  constructor(
+    private _url: string,
+    private _serverName: string,
+    private _canListUsers: boolean,
+    private _canListGroups: boolean,
+    private _canControlServer: boolean,
+    private _serverOwner: string,
+    private _hubApiUrl: string,
+    private _token: string,
+    private _trans: TranslationBundle
+  ) {
+    super();
+    this.populateBody(this.node);
+    this.addClass('jp-shared-link-body');
+    this.loadShares().then(() => {
+      this.updateSharesList().then(() => {
+        this.loadUsers();
+      });
+    });
+  }
+
+  /**
+   * Returns the input value.
+   */
+  getValue(): string {
+    return this._url;
+  }
+
+  protected onAfterAttach(msg: Message): void {
+    super.onAfterAttach(msg);
+  }
+
+  protected onBeforeDetach(msg: Message): void {
+    super.onBeforeDetach(msg);
+  }
+
+  private async loadUsers(): Promise<void> {
+    // If possible, download the users list for the UI
+    if (this._canListUsers) {
+      let offset = 0;
+      const limit = 200;
+      let usersData: any[] = [];
```

Comment:
We should avoid adding `any` in new code. If there is no interface for user already, let's define a simple one with `name` and whatever else is needed.